src/videojs-fairplay.js

```diff
@@ -0,0 +1,239 @@
+/* global videojs, WebKitMediaKeys */
+
+import * as util from './util';
+import concatInitDataIdAndCertificate from './fairplay';
+
+const {
+  arrayToString,
+  getHostnameFromURI,
+} = util;
+
+let logToBrowserConsole = false;
+
+class Html5Fairplay {
+  constructor(source, tech, options) {
+    options = options || tech.options_;
+
+    if (!source.src) {
+      return;
+    }
+
+    this.el_ = tech.el();
+    this.player_ = videojs(options.playerId);
+    this.protection_ = source && source.protection;
+    this.tech_ = tech;
+
+    this.onCertificateError = this.onCertificateError.bind(this);
+    this.onCertificateLoad = this.onCertificateLoad.bind(this);
+    this.onKeySessionWebkitKeyAdded = this.onKeySessionWebkitKeyAdded.bind(this);
+    this.onKeySessionWebkitKeyError = this.onKeySessionWebkitKeyError.bind(this);
+    this.onKeySessionWebkitKeyMessage = this.onKeySessionWebkitKeyMessage.bind(this);
+    this.onLicenseError = this.onLicenseError.bind(this);
+    this.onLicenseLoad = this.onLicenseLoad.bind(this);
+    this.onVideoError = this.onVideoError.bind(this);
+    this.onVideoWebkitNeedKey = this.onVideoWebkitNeedKey.bind(this);
+
+    tech.isReady_ = false;
+
+    this.src(source);
+
+    tech.triggerReady();
+  }
+
+  createKeySession(keySystem, initData) {
+    if (!this.el_.webkitKeys) {
+      if (WebKitMediaKeys.isTypeSupported(keySystem, 'video/mp4')) {
+        this.el_.webkitSetMediaKeys(new WebKitMediaKeys(keySystem));
+      } else {
+        throw new Error('Key System not supported');
+      }
+    }
+
+    if (!this.el_.webkitKeys) {
+      throw new Error('Could not create MediaKeys');
+    }
+
+    const keySession = this.el_.webkitKeys.createSession('video/mp4', initData);
+
+    if (!keySession) {
+      throw new Error('Could not create key session');
+    }
+
+    return keySession;
+  }
+
+  fetchCertificate({ callback }) {
+    const { certificateUrl } = this.protection_;
+
+    const request = new XMLHttpRequest();
+
+    request.responseType = 'arraybuffer';
+
+    request.addEventListener('error', this.onCertificateError, false);
+    request.addEventListener('load', (event) => {
+      this.onCertificateLoad(event, {
```

Comment:
Why is the 2nd param here an object with a `callback` inside of it?  It seems that we could just pass the callback in directly.  I didn't see any places that called `onCertificateLoad` with anything else in that 2nd option.